keps/sig-network/20190603-EndpointSlice-API.md

```diff
@@ -59,81 +59,116 @@ type EndpointSlice struct {
+    // addresses of this endpoint. The contents of this field are interpreted
+    // according to the corresponding EndpointSlice addressType field. This
+    // allows for cases like dual-stack (IPv4 and IPv6) networking. Consumers
+    // (e.g. kube-proxy) must handle different types of addresses in the context
+    // of their own capabilities. This must contain at least one address but no
+    // more than 100.
     // +listType=set
-    Targets []string `json:"targets,omitempty" protobuf:"bytes,1,opt,name=targets"`
-    // The Hostname of this endpoint.
-    // +optional 
-    Hostname *string `json:"hostname,omitempty" protobuf:"bytes,2,opt,name=hostname"`
-    // Required: the conditions of the endpoint.
-    Conditions EndpointConditions  `json:"conditions,omitempty" protobuf:"bytes,3,opt,name=conditions"`
-    // Reference to object providing the endpoint.
-    // +optional 
-    TargetRef *v1.ObjectReference `json:"targetRef,omitempty" protobuf:"bytes,4,opt,name=targetRef"`
-    // Topology can contain arbitrary topology information associated with the
+    Addresses []string `json:"addresses" protobuf:"bytes,1,rep,name=addresses"`
+    // conditions contains information about the current status of the endpoint.
+    Conditions EndpointConditions `json:"conditions,omitempty" protobuf:"bytes,2,opt,name=conditions"`
+    // hostname of this endpoint. This field may be used by consumers of
+    // endpoints to distinguish endpoints from each other (e.g. in DNS names).
+    // Multiple endpoints which use the same hostname should be considered
+    // fungible (e.g. multiple A values in DNS). Must pass DNS Label (RFC 1123)
+    // validation.
+    // +optional
+    Hostname *string `json:"hostname,omitempty" protobuf:"bytes,3,opt,name=hostname"`
+    // targetRef is a reference to a Kubernetes object that represents this
     // endpoint.
-    // Key/value pairs contained in topology must conform with the label format.
+    // +optional
+    TargetRef *v1.ObjectReference `json:"targetRef,omitempty" protobuf:"bytes,4,opt,name=targetRef"`
+    // topology contains arbitrary topology information associated with the
+    // endpoint. These key/value pairs must conform with the label format.
     // https://kubernetes.io/docs/concepts/overview/working-with-objects/labels
-    // Topology may contain the following well known keys:
-    // kubernetes.io/hostname: the value indicates the hostname of the node
-    // where the endpoint is located. This should match the corresponding
-    // node label.
-    // topology.kubernetes.io/zone: the value indicates the zone where the
-    // endpoint is located. This should match the corresponding node label.
-    // topology.kubernetes.io/region: the value indicates the region where the
-    // endpoint is located. This should match the corresponding node label.
-    // +optional 
+    // Topology may include a maximum of 16 key/value pairs. This includes, but
```

Comment:
"For endpoint backed by k8s pod, this may include"

Reply:
Done, thanks!